tracker/server.go

```diff
@@ -0,0 +1,256 @@
+package tracker
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/R-a-dio/valkyrie/config"
+	"github.com/R-a-dio/valkyrie/errors"
+	"github.com/R-a-dio/valkyrie/rpc"
+	"github.com/R-a-dio/valkyrie/util"
+	"github.com/R-a-dio/valkyrie/website"
+	"github.com/Wessie/fdstore"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/hlog"
+	"google.golang.org/grpc"
+)
+
+const (
+	// UpdateListenersTickrate is the period between two UpdateListeners
+	// calls done to the manager
+	UpdateListenersTickrate = time.Second * 10
+	// SyncListenersTickrate is the period between two sync operations
+	SyncListenersTickrate = time.Minute * 10
+
+	RemoveStaleTickrate = time.Hour * 24
+	RemoveStalePeriod   = time.Minute * 5
+
+	HttpLn      = "tracker.http"
+	GrpcLn      = "tracker.grpc"
+	TrackerFile = "tracker.listeners"
+
+	ICECAST_AUTH_HEADER         = "icecast-auth-user"
+	ICECAST_CLIENTID_FIELD_NAME = "client"
+)
+
+type Server struct {
+	cfg config.Config
+
+	recorder *Recorder
+	http     *http.Server
+	grpc     *grpc.Server
+	httpLn   net.Listener
+	grpcLn   net.Listener
+
+	h http.Handler
+}
+
+func (s *Server) Start(ctx context.Context, fds *fdstore.Store) error {
```

Comment:
Can move this functions content back into execute and use the `httpLn` data slot for storing the recorder state